spec/requests/form_odk_rendering_spec.rb

```diff
@@ -1,82 +1,98 @@
+      @form = create(:form, question_types: %w(text long_text integer decimal location select_one
+        multilevel_select_one select_multiple datetime date time))
       @form.publish!
       get(form_path(@form, format: :xml))
     end
 
-    it 'should render proper xml' do
+    it "should render proper xml" do
       expect(response).to be_success
 
       # Parse the XML and tidy.
       doc = Nokogiri::XML(response.body) { |config| config.noblanks }
-      expect(doc.to_s).to eq File.read(File.expand_path('../../expectations/odk/sample_form.xml', __FILE__))
+      expect(doc.to_s).to eq expectation_file("odk/sample_form.xml")
     end
   end
 
-  context 'grid form' do
+  context "grid form" do
     before do
-      first_question = create(:question, qtype_name: 'select_one')
-      second_question = create(:question, qtype_name: 'select_one', option_set: first_question.option_set)
+      first_question = create(:question, qtype_name: "select_one")
+      second_question = create(:question, qtype_name: "select_one", option_set: first_question.option_set)
 
       @form = create(:form, questions: [[first_question, second_question]])
       @form.publish!
       get(form_path(@form, format: :xml))
     end
 
-    it 'should render proper xml' do
+    it "should render proper xml" do
       expect(response).to be_success
 
       doc = Nokogiri::XML(response.body) { |config| config.noblanks }
-      expect(doc.to_s).to eq File.read(File.expand_path('../../expectations/odk/grid_form.xml', __FILE__))
+      expect(doc.to_s).to eq expectation_file("odk/grid_form.xml")
     end
   end
 
-  context 'form with & in option name' do
+  context "form with & in option name" do
     before do
       @form = create(:form, question_types: %w(select_one))
-      @option_set = create(:option_set, option_names: ['Salt & Pepper', 'Peanut Butter & Jelly'])
+      @option_set = create(:option_set, option_names: ["Salt & Pepper", "Peanut Butter & Jelly"])
       @form.questions.first.update_attributes!(option_set: @option_set)
       @form.publish!
       get(form_path(@form, format: :xml))
     end
 
-    it 'should not have parsing errors' do
+    it "should not have parsing errors" do
       expect(response).to be_success
       doc = Nokogiri::XML(response.body) { |config| config.noblanks }
       expect(doc.errors).to be_empty
     end
   end
 
-  context 'group form' do
+  context "media question form" do
     before do
-      @form = create(:form, question_types: [['text', 'text', 'text', 'text']])
+      @form = create(:form, question_types: %w(text image annotated_image sketch signature audio video))
+      @form.publish!
+      get(form_path(@form, format: :xml))
+    end
+
+    it "should render proper XML" do
+      expect(response).to be_success
+
+      doc = Nokogiri::XML(response.body) { |config| config.noblanks }
+      expect(doc.to_s).to eq expectation_file("odk/media_question_form.xml")
```

Comment:
The above two lines are repeated in quite a few places here. Perhaps move into a custom matcher or helper method?